kubernetes.go

```diff
@@ -128,13 +130,79 @@ func homeDir() string {
 	return os.Getenv("USERPROFILE") // windows
 }
 
-func kubeGetReadinessHandler(ignoreDaemonSets bool) (readiness, error) {
+func kubeGetReadinessHandler(ignoreDaemonSets bool, deleteLocalData bool) (readiness, error) {
 	clientset, err := kubeGetClientset()
 	if err != nil {
 		log.Fatalf("Error getting kubernetes connection: %v", err)
 	}
 	if clientset == nil {
 		return nil, nil
 	}
-	return &kubernetesReadiness{clientset: clientset, ignoreDaemonSets: ignoreDaemonSets}, nil
+	return &kubernetesReadiness{clientset: clientset, ignoreDaemonSets: ignoreDaemonSets, deleteLocalData: deleteLocalData}, nil
+}
+
+// setScaleDownDisabledAnnotation set the "cluster-autoscaler.kubernetes.io/scale-down-disabled" annotation
+// on the list of nodes if required. Returns a list of hostname where the annotation
+// is applied.
+func setScaleDownDisabledAnnotation(hostnames []string) ([]string, error) {
+	// get the node reference - first need the hostname
+	var (
+		node      *corev1.Node
+		hostname  string
+		err       error
+		key       = "cluster-autoscaler.kubernetes.io/scale-down-disabled"
+		annotated = []string{}
+	)
+	clientset, err := kubeGetClientset()
+	if err != nil {
+		log.Fatalf("Error getting kubernetes connection: %v", err)
+	}
+	if clientset == nil {
+		return annotated, nil
+	}
+	nodes := clientset.CoreV1().Nodes()
+	for _, h := range hostnames {
+		node, err = nodes.Get(h, v1.GetOptions{})
+		if err != nil {
+			return annotated, fmt.Errorf("Unexpected error getting kubernetes node %s: %v", hostname, err)
+		}
+		annotations := node.GetAnnotations()
```

Comment:
This always struck me as silly, that you have to set all annotations at once, rather than being able to set a single one. Oh well.

Reply:
Never really looked at the k8s code, but my best guess is that all the annotations are saved as a blob in etcd, in contrast to label.